src/Compilers/Core/MSBuildTask/DiscoverEditorConfigFiles.cs

```diff
@@ -0,0 +1,221 @@
+                                if (matchesCaseInsensitiveAscii(line, charIndex, root))
+                                {
+                                    charIndex += root.Length;
+                                    charIndex = skipSpaces(line, charIndex);
+                                    if (charIndex < line.Length)
+                                    {
+                                        switch (line[charIndex])
+                                        {
+                                            case '=':
+                                            case ':':
+                                                charIndex = skipSpaces(line, charIndex + 1);
+                                                if (matchesCaseInsensitiveAscii(line, charIndex, trueLiteral))
+                                                {
+                                                    charIndex += trueLiteral.Length;
+                                                    charIndex = skipSpaces(line, charIndex);
+                                                    // Is the rest of the line empty or a comment?
+                                                    if (charIndex == line.Length ||
+                                                        line[charIndex] == '#' ||
+                                                        line[charIndex] == ';')
+                                                    {
+                                                        return true;
+                                                    }
+                                                }
+                                                break;
+                                        }
+                                    }
+                                }
+                                break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+
+            return false;
+
+            int skipSpaces(string str, int startIndex)
+            {
+                for (int cur = startIndex; cur < str.Length; cur++)
+                {
+                    if (!char.IsWhiteSpace(str[cur]))
+                    {
+                        return cur;
+                    }
+                }
+
+                return str.Length;
+            }
+
+            bool matchesCaseInsensitiveAscii(string str, int strIndex, string targetLower)
+            {
+                if (strIndex + targetLower.Length > str.Length)
+                {
+                    return false;
+                }
+
+                for (int targetIndex = 0; targetIndex < targetLower.Length; targetIndex++)
+                {
+                    char curChar = str[strIndex + targetIndex];
+                    char targetChar = targetLower[targetIndex];
+                    if (curChar != targetChar && curChar != (targetChar - 32))
```

Comment:
>32 [](start = 74, length = 2)

Can you please make this `32` less magical?  Like maybe 'A'-'a' (or is it the other way around)